enhancements/api-review/custom-tags-aws.md

```diff
@@ -79,65 +101,373 @@ The userTags field is intended to be set at install time and is considered immut
 
 ### Graduation Criteria
 
-TBD
-
 #### Dev Preview -> Tech Preview
 
-TBD
+NA
 
 #### Tech Preview -> GA
 
-TBD
+- Upgrade/downgrade testing
+- Sufficient time for feedback
+- Available by default
+- Stress testing for scaling and tag update scenarios
 
 #### Removing a deprecated feature
 
-This enhancement does not remove or deprecate any features.
+`experimentalPropagateUserTags` field will be set for deprecation.
+`.status.platformStatus.aws` will be set for deprecation.
 
 ### Upgrade / Downgrade Strategy
 
 On upgrade:
 
-The new status field won't be populated since it is only populated by the installer and that can't have happened if the cluster was installed from a prior version. Components that consume the new field should take no action since they will see no additional tags.
+On upgrade from version supporting `experimentalPropagateUserTags`, openshift components that consume the new field should merge `.status.platformStatus.aws` and `.spec.platformSpec.aws`.
+`.spec.platformSpec.aws` can be updated or changed which will override the values from status.
+
+For installer configuration compatibility, configuration with `experimentalPropagateUserTags` should be supported.
 
 On downgrade:
 
-The status field may remain populated, components may or may not continue to tag newly created resources w/ the additional tags depending on whether or not a given component still has logic to respect the status tags, after the downgrade.
+The status/spec field may remain populated, components may or may not continue to tag newly created resources with the additional tags depending on whether or not a given component still has logic to respect the status tags, after the downgrade.
 
 ### Version Skew Strategy
 
-TBD
-
 ## Implementation History
 
 ## Drawbacks
 
-If a customer decides they do not want these tags applied to new resources, there is no clean way to address that need:
-
-1. they would have to get help from support to edit the status field
-2. they would have to manually remove the undesired tags from any existing resources
+If a user decides that they do not want these tags applied to new resources, there is no clean way to address that need:
 
-In the future we will want to introduce a spec field where a customer can specify and edit these tags, which will be reflected into status when changed, and we will expect consuming components to reconcile changes to the set of tags by applying net new tags to existing resources (but they will still not remove tags that are dropped from the list of tags).
+1. User would have to get help from support to edit the status field for older version which uses `experimentalPropagateUserTags`.
+2. User would have to manually remove the undesired user-defined tags from some existing resources even though user-defined tags can be marked for deletion by setting tag value to empty string.
+3. User would have to manually update user-defined tags for AWS resources which are not managed by an operator.
```

Comment:
I would like to see an overview of how we would expect a user to go about updating non-managed AWS resources. I expect that we are going to need to provide guidance in the docs anyway.